fvm/evm/debug/tracer.go

```diff
@@ -0,0 +1,76 @@
+package debug
+
+import (
+	"encoding/json"
+
+	gethCommon "github.com/onflow/go-ethereum/common"
+	"github.com/onflow/go-ethereum/eth/tracers"
+	"github.com/rs/zerolog"
+
+	// this import is needed for side-effects, because the
+	// tracers.DefaultDirectory is relying on the init function
+	_ "github.com/onflow/go-ethereum/eth/tracers/native"
+)
+
+type EVMTracer interface {
+	TxTracer() tracers.Tracer
+	Collect(id gethCommon.Hash)
+}
+
+var _ EVMTracer = &CallTracer{}
+
+type CallTracer struct {
+	logger   zerolog.Logger
+	tracer   tracers.Tracer
+	uploader Uploader
+}
+
+func NewEVMCallTracer(uploader Uploader, logger zerolog.Logger) (*CallTracer, error) {
+	tracerType := json.RawMessage(`{ "onlyTopCall": true }`)
+	tracer, err := tracers.DefaultDirectory.New("callTracer", &tracers.Context{}, tracerType)
```

Comment:
Could use some constants

Reply:
Yeah, will do that. Btw this is super ugly how they accept json raw message but it's just the way it is.